File: app/src/main/java/test/itexico/movies/view/EpisodesFragment.java
package test.itexico.movies.view;

import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.BindView;
import butterknife.ButterKnife;
import test.itexico.movies.R;
import test.itexico.movies.presenter.EpisodesListPresenter;

public class EpisodesFragment extends Fragment {

    @BindView(R.id.list_episodes) RecyclerView listEpisodes;
    @BindView(R.id.headerLayout) ConstraintLayout headerLayout;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.episodes_fragment, container, false);

        Bundle extras = getArguments();
        String seasonId = extras.getString(getResources().getString(R.string.key_sesionId));

        ButterKnife.bind(this, rootView);

        listEpisodes.setLayoutManager(new LinearLayoutManager(getContext()));

        EpisodesListPresenter episodesActivityPresenter = new EpisodesListPresenter(getContext(), headerLayout, listEpisodes);
        episodesActivityPresenter.setHeaderInfo(extras);

        if(Integer.valueOf(seasonId)<0){
            DialogAlert.show(getContext(),
                    getResources().getString(R.string.err_no_data_title),
                    getResources().getString(R.string.err_no_data_text),
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog, int which) {
                            getActivity().finish();
                        }
            });
        }else {
            episodesActivityPresenter.populateEpisodesFromSeason(Integer.valueOf(seasonId));
        }

        return rootView;
    }
}

File: app/src/main/java/test/itexico/movies/view/EpisodesFragment.kt
package test.itexico.movies.view

import android.arch.lifecycle.Lifecycle
import android.arch.lifecycle.LifecycleObserver
import android.arch.lifecycle.LifecycleOwner
import android.arch.lifecycle.ViewModelProviders
import android.content.DialogInterface
import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.android.volley.Response
import kotlinx.android.synthetic.main.episodes_fragment.view.*
import test.itexico.movies.R
import test.itexico.movies.model.EpisodesListModel
import test.itexico.movies.model.EpisodesListModel.*
import test.itexico.movies.presenter.EpisodesListPresenter

class EpisodesFragment : Fragment(), LifecycleObserver{

    internal var listEpisodes: RecyclerView? = null
    internal var headerLayout: ConstraintLayout? = null
    lateinit var episodesActivityPresenter: EpisodesListPresenter

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater?.inflate(R.layout.episodes_fragment, container, false)
        val seasonId = arguments.getInt(resources.getString(R.string.key_seasonNum))

        listEpisodes = rootView?.list_episodes
        headerLayout = rootView?.header_layout

        listEpisodes?.layoutManager = LinearLayoutManager(context)

        episodesActivityPresenter = EpisodesListPresenter(context, headerLayout as ConstraintLayout, listEpisodes as RecyclerView)
        episodesActivityPresenter.setHeaderInfo(arguments)

        if (seasonId < 0) {
            DialogAlert.show(context,
                    resources.getString(R.string.err_no_data_title),
                    resources.getString(R.string.err_no_data_text),
                    DialogInterface.OnClickListener { _, _ -> activity.finish() })
        }

        return rootView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (lifecycle.currentState.isAtLeast(Lifecycle.State.CREATED)) {
            val seasonId = arguments.getInt(resources.getString(R.string.key_seasonNum))
            val viewModel = ViewModelProviders.of(this,
                    EpisodesListModelFactory(
                            this.activity.application,
                            seasonId,
                            Response.ErrorListener { error ->
                                episodesActivityPresenter.onErrorResponse(error)
                            }))
                    .get(EpisodesListModel::class.java)
            viewModel.getData().observe(this, android.arch.lifecycle.Observer { results ->
                episodesActivityPresenter.setData(results!!)
            })
        }
    }
}
